lib/logstash/inputs/rabbitmq/march_hare.rb

```diff
@@ -100,6 +102,9 @@ def setup
 
       # exchange binding is optional for the input
       if @exchange
+        @ch.exchange(@exchange,
```

Comment:
Some may find it convenient to have Logstash pre-declare their topology. There is no need to declare an exchange to consume from a queue. However, if you do, then make sure you provide exchange type, because it defines how routing is done and thus is crucially important.